clang/test/CXX/temp/temp.spec/temp.expl.spec/p20-2.cpp

```diff
@@ -4,7 +4,8 @@ void f(T);
 
 template<typename T>
 struct A {
-  // expected-error@+1{{cannot declare an explicit specialization in a friend}}
+  // expected-error@+2{{cannot declare an explicit specialization in a friend}}
+  // expected-error@+1{{friend function specialization cannot be defined}}
```

Comment:
We probably shouldn't be double-erroring on all of these.  In this case, the existing error seems preferential.

Reply:
@erichkeane I suppose we don't want double-erroring for any of these https://godbolt.org/z/Wdo7bqe1v... if that is the case, perhaps we should postpone diagnosing friend function definitions until after we call `ActOnFunctionDecl`?